tests/installation/partitioning_togglehome.pm

```diff
@@ -31,16 +31,21 @@ sub run {
     assert_screen [qw(inst-partition-radio-buttons enabledhome disabledhome)];
     # For s390x there was no offering of separated home partition until SLE 15 See bsc#1072869
     if ((!check_var('ARCH', 's390x') or is_storage_ng()) and !match_has_tag('disabledhome')) {
-        # older versions have radio buttons and no separate swap block
-        # For SLE 15 SP1 we got Cancel button instead of Abort, so have new shortcut now
-        # Expecting it for TW and Leap 15.1, but not yet there
-        if (match_has_tag('inst-partition-radio-buttons') || is_sle('15-SP1+')) {
-            $cmd{toggle_home} = 'alt-r';
-        }    # Have different shortkey on storage-ng without LVM
-        elsif (is_storage_ng && !get_var('LVM')) {
-            $cmd{toggle_home} = 'alt-o';
+
+        if (check_var('VIDEOMODE', 'text') || match_has_tag('inst-partition-radio-buttons')) {
+            # older versions have radio buttons and no separate swap block
+            # For SLE 15 SP1 we got Cancel button instead of Abort, so have new shortcut now
+            # Expecting it for TW and Leap 15.1, but not yet there
+            if (match_has_tag('inst-partition-radio-buttons')) {
+                $cmd{toggle_home} = 'alt-r';
+            }    # Have different shortkey on storage-ng without LVM
+            elsif (is_storage_ng && !get_var('LVM')) {
+                $cmd{toggle_home} = 'alt-o';
+            }
+            send_key $cmd{toggle_home};
+        } else {
+            assert_and_click 'enabledhome';
```

Comment:
Hm, what value do you see in needing to maintain the shortcuts for ncurses plus additionally the needles? I personally prefer the old approach.

Reply:
We don't introduce any new needle, so there is no extra work. Old approach showed that it doesn't work, yast UI framework is the way out.

Reply:
well, I am not yet sure about that but the change is ok nevertheless :)

Reply:
You are not sure about new needles? I've reused both tags, as for other approaches I didn't hear any proposals from you, but I'm very interested.

Reply:
Because you stated that you are interested I will try to explain what I mean: You did not introduce new needle tags, that's good from test maintenance point of view. However, we still need to maintain the needles for the graphical mode but also the mapping of hotkeys to keypresses for the non-graphical mode. So I am not sure this approach is less test maintenance work then just relying on hotkeys + keypresses.

Reply:
For openSUSE we have different shortcut, but no installation in textmode where we disable home, so it's already less work. For textmode we still need both needles to assert state. But thanks, got your point now.